lib/sdf-server/src/service/v2/workspace/list_workspace_users.rs

```diff
@@ -0,0 +1,47 @@
+use axum::{extract::Path, Json};
+use dal::{UserPk, WorkspacePk};
+use serde::{Deserialize, Serialize};
+
+use crate::{extract::HandlerContext, service::v2::AccessBuilder};
+
+use super::WorkspaceAPIError;
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct User {
+    pub id: UserPk,
+    pub name: String,
+    pub email: String,
+}
+
+impl From<dal::User> for User {
+    fn from(value: dal::User) -> Self {
+        Self {
+            id: value.pk(),
+            name: value.name().to_owned(),
+            email: value.email().to_owned(),
+        }
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq, Eq)]
+#[serde(rename_all = "camelCase")]
+pub struct Response {
+    users: Vec<User>,
+}
+
+pub async fn list_workspace_users(
+    HandlerContext(builder): HandlerContext,
+    AccessBuilder(access_builder): AccessBuilder,
+    Path(workspace_id): Path<WorkspacePk>,
+) -> Result<Json<Response>, WorkspaceAPIError> {
+    let ctx = builder.build_head(access_builder).await?;
```

Comment:
Like before, the context is built using the HEAD change set, which we may or may not want in the long term. However, this reflects existing behavior minus the requirement for an "admin" DalContext.

Reply:
FYI, for future endpoints under v2, you can remove the Path() from this entirely and use `ctx.workspace_id()`, which already gets it from path. I didn't make this change for most of the endpoints simply because I wanted that churn to happen separately in a refactor PR (and my authentication PR was already big enough). What you are doing right now is perfectly in line with all the rest of v2.